source/MaterialXCore/Definition.cpp

```diff
@@ -36,22 +36,34 @@ InterfaceElementPtr NodeDef::getImplementation(const string& target, const strin
     vector<InterfaceElementPtr> interfaces = getDocument()->getMatchingImplementations(getQualifiedName(getName()));
     vector<InterfaceElementPtr> secondary = getDocument()->getMatchingImplementations(getName());
     interfaces.insert(interfaces.end(), secondary.begin(), secondary.end());
+
+    // Scan for implementations, looking for those with a language match 
+    // over those which do not.
+    bool matchLanguage = !language.empty();
     for (InterfaceElementPtr interface : interfaces)
     {
-        if (!targetStringsMatch(interface->getTarget(), target) ||
+        ImplementationPtr implement = interface->asA<Implementation>();
+        if (!implement||
+            !targetStringsMatch(interface->getTarget(), target) ||
             !isVersionCompatible(interface))
         {
             continue;
         }
-        if (!language.empty())
+        if (!matchLanguage || 
+            implement->getLanguage() == language)
         {
-            // If the given interface is an implementation element, as opposed to
-            // a node graph, then check for a language string match.
-            ImplementationPtr implement = interface->asA<Implementation>();
-            if (implement && implement->getLanguage() != language)
-            {
-                continue;
-            }
+            return interface;
+        }
+    }
+
+    // Search for a nodegraph match if no implementation match was found
+    for (InterfaceElementPtr interface : interfaces)
```

Comment:
nodegraph scan. Find the first one and return it as nodegraphs are not language specific.